### PR TITLE
Use docker hub

### PR DESCRIPTION
**Pull request by soerenmartius**
This is a conceptual PR with that contains ongoing work and isn't finished or ready for review yet. Just asking for early feedback.

I moved the Dockerfile to a new [build-tools](https://github.com/mineiros-io/build-tools) repository. Also, I've set up the repository in [docker hub](https://hub.docker.com/r/mineiros/build-tools) and enabled auto-build also. The Makefile will now pull the image from [hub.docker.com](https://hub.docker.com/r/mineiros/build-tools) instead of building it locally.

**To-Do:**
- Move the generic Makefile targets to the build-tools repository and load them in each repositories Makefile individually.
- Think about security. ( unprivileged user in Dockerfile, Mount Volume as Read-Only )
- We should inject the plan file from the `docker-run-terraform-plan`  command to the `docker-run-terraform-apply` command. At the moment, `docker-run-terraform-apply` creates it's own state file.
- If we load the generic Makefile targets from the build-tools repository, how can we prevent MITM attacks?

### CODE REVIEW

**Comment by soerenmartius** [comment]
ok, that works as expected already. The next step will be to move the `Makefile` to the [build-tools](https://github.com/mineiros-io/build-tools) repository and to include in the modules repository with curl.

The biggest question remains though: Can we get this to work on a windows powershell 👎

File: Makefile [hunk 1]
```diff
@@ -1,78 +1,51 @@
-#!/usr/bin/env make
+MOUNT_TARGET_DIRECTORY  = /app/src
+BUILD_TOOLS_DOCKER_REPO = mineiros/build-tools
+DOCKER                  = $(shell which docker)
```
Comment (mariux) [inline Makefile]: this is a noop....? so you can use `docker` right away it willl fail no matter what?

File: Makefile [hunk 2]
```diff
@@ -1,78 +1,51 @@
...
+				gsub(":+$$", "", cmd); \
+				printf "  \x1b[32;01m%-35s\x1b[0m %s\n", cmd, msg; \
+			} \
+	} \
+	{ lastLine = $$0 }' $(MAKEFILE_LIST) | sort -u
+
+.DEFAULT_GOAL := help
+
+## Mounts the working directory inside a docker container and runs the pre-commit hooks
+docker-run-pre-commit-hooks: $(DOCKER)
```
Comment (mariux) [inline Makefile]: should not depend on the binary..

Comment (soerenmartius) [reply Makefile]: ok, I've changed that

File: Makefile (hunk 1 above)
Comment (soerenmartius) [reply Makefile]: ok, I've changed that

**Comment by soerenmartius** [comment]
rdy for final review @mariux

**Pull request merged**